aten/src/ATen/Half.h

```diff
@@ -10,6 +10,15 @@
 #include <cuda_fp16.h>
 #endif
 
+#ifdef _MSC_VER
+// MSVC does not support std::inf and std::nan for integral types
+#define ISINF(x) std::isinf((double)x)
+#define ISNAN(x) std::isnan((double)x)
```

Comment:
I think a better fix would be to always return `false` for integral types (should be possible to do inside this macro with some template magic). Not all values of `long long` can be represented in `double`